graphdatascience/graph/graph_type_check.py

```diff
@@ -0,0 +1,48 @@
+from functools import wraps
+from typing import Any, Callable, Optional, TypeVar, cast
+
+from .graph_object import Graph
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+def graph_type_check(func: F) -> F:
```

Comment:
We don't accept strings since we don't want to encourage people to pass around strings in their code. So having to use the `Graph` object will improve their code, and at the same time expose the nice features of the object. That's the reasoning I believe

Reply:
maybe the encouragement could be done via warning as well.

Reply:
true :)